src/Features/Customer/UserProfile/Profile.js

```diff
@@ -1,162 +1,148 @@
       errorMessage: '',
-
-      onChange: (event, index) => {
-        setEmail(event.target.value);
-
-        fields[index].value = event.target.value;
-
-        fields[index].getValidation(event.target.value, index);
+      onChange: ({ target: { value } }, index) => {
+        const updatedFields = fieldChangeHandler(fields, value, index);
+        setFields(updatedFields);
       },
-
-      getValidation: (value, index) => {
+      getValidation: (value) => {
         if (!emailRegex.test(value)) {
-          fields[index].errorMessage = 'Email type is not valid';
-
-          fields[index].isValid = false;
-        } else {
-          fields[index].errorMessage = '';
-
-          fields[index].isValid = true;
+          return 'Email type is not valid';
         }
+        return '';
+      },
+    },
+    {
+      type: TEXT_FIELD,
+      textFieldType: 'password',
+      label: 'Password',
+      variant: 'standard',
+      value: '',
+      errorMessage: '',
+      onChange: ({ target: { value } }, index) => {
+        const updatedFields = fieldChangeHandler(fields, value, index);
+        setFields(updatedFields);
+      },
+      getValidation: (value) => {
+        if (value.length < 8) {
+          return 'Password must be 8 characters long';
+        }
+        return '';
       },
     },
-
     {
       type: TEXT_FIELD,
-
       textFieldType: 'text',
-
       label: 'Contact',
-
       variant: 'standard',
-
-      value: '03412132212',
-      disabled: isEdit,
-      isValid: true,
-
+      value: '',
       errorMessage: '',
-
-      onChange: (event, index) => {
-        setContact(event.target.value);
-
-        fields[index].value = event.target.value;
-
-        fields[index].getValidation(event.target.value, index);
+      onChange: ({ target: { value } }, index) => {
+        const updatedFields = fieldChangeHandler(fields, value, index);
+        setFields(updatedFields);
       },
-
-      getValidation: (value, index) => {
+      getValidation: (value) => {
         if (!contactRegex.test(value)) {
-          fields[index].errorMessage = 'Contact length or Type is not valid';
-
-          fields[index].isValid = false;
-        } else {
-          fields[index].errorMessage = '';
-
-          fields[index].isValid = true;
+          return 'Contact length or Type is not valid';
         }
+        return '';
       },
     },
   ]);
 
   const saveHandler = () => {
-    validateOnSubmit() ? setOnSaveSuccess(true) : setOnSaveSuccess(false);
+    const { validateArray, isValid } = validateOnSubmit(fields);
+    setFields(validateArray);
+    const [{ value: name }, { value: email }, { value: password }] = fields;
+
+    isValid &&
+      editUserMutate({
+        id,
+        body: {
+          name,
+          email,
+          password,
+        },
+        token,
+      });
   };
 
   const buttons = {
     button: [
       {
         type: 'button',
-
-        name: 'Edit',
-
+        name: 'save',
```

Comment:
I think there should be condition to check if it is save screen button or edit ??